qiskit_experiments/framework/experiment_data.py

```diff
@@ -631,27 +632,34 @@ def _set_backend(self, new_backend: Backend, recursive: bool = True) -> None:
 
     def _set_hgp_from_provider(self, provider):
         try:
-            hub = None
-            group = None
-            project = None
             # qiskit-ibmq-provider style
             if hasattr(provider, "credentials"):
                 creds = provider.credentials
-                hub = creds.hub
-                group = creds.group
-                project = creds.project
+                self.hgp = f"{creds.hub}/{creds.group}/{creds.project}"
```

Comment:
Do all providers supply non-None hub/group/project in their credentials?

Reply:
I think so; this code can fail (it's in a try clause) so we cover the case where they do not.